eclipse-theia/theia
# plugin-ext-metrics extension for creating metrics from language plugins

#### What it does

This PR introduces a new extension that creates metrics from language plugins. It creates metrics for two specific instances:
- The time it takes for a language server request to be made for a specific request for a specific plugin
E.g. vscode-yaml -> document symbols -> 13 ms

- The percentage of success made for a specific request for a specific plugin
E.g. vscode-yaml -> document symbols -> 99% successful

Spectrum discussion: https://spectrum.chat/theia/dev/current-state-of-metrics-in-theia~8efd05d2-a25b-4629-a1b1-1dd8a0ccef96
Related issues:
https://github.com/eclipse/che/issues/14245

#### How to test


For testing I used a modified version of vscode-yaml that errors randomly in document symbols (just so you don't have to go out and find a plugin that errors on a language server request): you can download from https://github.com/JPinkney/Sample-vsix-repo/blob/master/0.5.2-beta7.vsix

You can actually use anything in this list though: https://github.com/eclipse-theia/theia/compare/master...JPinkney:plugin-ext-metrics?expand=1#diff-fb56011c00b860bfb7d05eedd825006eR33 it's just easier to verify that the percentage of successes is working when you know how to generate an error in the plugin.

Play around with the extension (autocompletion, document symbols, etc, etc) just so that we can get some data for metrics.

After you've played around with it go to ${yourUrl}/metrics and you should see the updated analytics for language_server_success_metrics and language_server_time_metrics.


#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
